### PR TITLE
sys/shell/commands/ping: fix dependency & convert to ztimer

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

Add missing dependency to xtimer so that the shell command `ping` is
again provided when requested.

### Testing procedure

Run `make BOARD=nucleo-f767zi -C examples/gnrc_networking flash term` and then `help`. If the `ping` command is listed again, the bug is fixed. (Did so, worked for me.)

### Issues/PRs references

None

### CODE REVIEW

**Comment by maribu** [comment]
I also added a commit to convert to `ztimer` on top of the fix.

@chrysn: IMO the fix commit would be backport material. The conversion to ztimer can IMO wait for the next release.

**Comment by chrysn** [comment]
Indeed (it's not part of the test suite but easily verified).

File: sys/Makefile.dep
Comment (miri64) [inline sys/Makefile.dep:318]: I don't like this. The code of `gnrc_icmpv6_echo` does not require `ztimer`, but the shell command which implements `ping` does (and that is how the dependency used to be expressed). Pulling in `ztimer_usec` here as a dependency can have unintentional repercussions for testing, so I rather prefer to keep the dependency as is: Pull in `sc_gnrc_icmpv6_echo.c`, when a timer module is present.

File: sys/shell/commands/Makefile
Comment (miri64) [inline sys/shell/commands/Makefile:69]: So check for `ztimer_usec` here instead of (or in addition to) `xtimer`.

File: sys/shell/commands/sc_gnrc_icmpv6_echo.c
Comment (miri64) [inline sys/shell/commands/sc_gnrc_icmpv6_echo.c:113]: I know this is meant to be a simple bug fix and not a `ztimer` conversion, but I don't think, the ping module actually needs microsecond precision (the final result is printed in milliseconds). And I am a bit afraid, once this is merged, it will be stuck with microseconds, because nobody cares about converting to the (potentially more energy saving) milliseconds timer. If there is no API change resulting from that, would it make more sense to directly use `ZTIMER_MSEC` here?

File: sys/Makefile.dep
Comment (miri64) [reply sys/Makefile.dep:318]: Also take note that `gnrc_icmpv6_echo` is also used for the "ICMPv6 ping server" side, where no notion of time is needed => a possibly unneeded timer module is pulled in.

Comment (maribu) [reply sys/Makefile.dep:318]: Note that this is all inside

```Makefile
ifneq (,$(filter shell_commands,$(USEMODULE)))
  [...]
endif
```

if the user wants shell commands, the user should get shell commands. (And adding a fine-grained selection of shell commands would IMO be a welcome addition, but is not the goal of this PR.)

Comment (miri64) [reply sys/Makefile.dep:318]: Ah, sorry I missed that context!

Comment (miri64) [reply sys/Makefile.dep:318]: Then all good.

File: sys/shell/commands/sc_gnrc_icmpv6_echo.c
Comment (benpicco) [reply sys/shell/commands/sc_gnrc_icmpv6_echo.c:113]: I very much like to do fast pings and the result is already printed using µs accuracy

```
> ping ff02::1

12 bytes from fe80::ac59:5aff:fe33:896e%6: icmp_seq=0 ttl=64 time=0.161 ms
12 bytes from fe80::ac59:5aff:fe33:896e%6: icmp_seq=1 ttl=64 time=0.167 ms
12 bytes from fe80::ac59:5aff:fe33:896e%6: icmp_seq=2 ttl=64 time=0.186 ms

--- ff02::1 PING statistics ---
3 packets transmitted, 3 packets received, 0% packet loss
round-trip min/avg/max = 0.161/0.171/0.186 ms
```

I'm also not so worried about the energy consumption of a shell command.

**Review by kfessel — COMMENTED** [review]
seems like we are missing a `gnrc_ping_cmd` (pseudo)module -in other words why should a icmp-echo-service also add the client part.

**Review by miri64 — APPROVED** [review]
Considering that this is meant to be backported and should as thus be as minimal as possible: ACK. Adding a new (pseudo-)module to specifically select the `ping` command can be done as a follow-up.

**Comment by maribu** [comment]
I'd much rather see making all shell commands submodules :)

**Comment by maribu** [comment]
Thx :)

**Comment by miri64** [comment]
> Indeed (it's not part of the test suite but easily verified).

Please provide a backport.

**Comment by maribu** [comment]
Backport provided in #18354

**Pull request merged**
